tests/rustdoc/book-location.rs

```diff
@@ -0,0 +1,7 @@
+//@ compile-flags: -Zunstable-options --book-location https://somewhere.world
```

Comment:
We should have another (run-make) test that tests the mdbook integration, especially since that's where most of the complexity of this PR is.